src/app/auth.guard.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, ActivatedRouteSnapshot, RouterStateSnapshot, Router } from '@angular/router';
+import { Observable } from 'rxjs';
+import { select, Store } from '@ngrx/store';
+import * as UserSelectors from './header/state/user/user.selectors';
+import {map} from "rxjs/operators";
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthGuard implements CanActivate {
+
+  constructor(private store: Store, private router: Router) {}
+
+  canActivate(
+    next: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot
+  ): Observable<boolean> | Promise<boolean> | boolean {
+    return this.store.pipe(
+      select(UserSelectors.selectIsCreated),
+      map((isUserCreated) => {
+        if (isUserCreated) {
+          return true;
+        } else {
+          this.router.navigate(['/']);
```

Comment:
fyi: did you know this is a promise ? usually it's not awaited (fire and forget) ... it could be prefixed with "void" to make it more clear 
```ts
  void this router.navigate([]);
```

Reply:
i thought to use .then but it didn't look well to me, thank you, I will use void

Reply:
as i said .. it's no hard requirement :-) just wrote it for your interest